src/test/java/core/basesyntax/CalculatorTest.java

```diff
@@ -0,0 +1,109 @@
+package core.basesyntax;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class CalculatorTest {
+    private static final double DELTA = 1E-10;
+    private static final double FIRST_POSITIVE = 12;
+    private static final double FIRST_NEGATIVE = -3;
+    private static final double SECOND_POSITIVE = 6;
+    private static final double SECOND_NEGATIVE = -4;
+    private static final double EVEN_POSITIVE = 2;
+    private static final double ODD_POSITIVE = 3;
+    private static final double EVEN_NEGATIVE = -2;
+    private static final double ODD_NEGATIVE = -3;
+    private static final double FRACTION = 0.5;
+    private Calculator calculator;
+
+    @Before
+    public void setUp() {
+        calculator = new Calculator();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void nonexistentOperation() {
+        calculator.calculate(FIRST_POSITIVE, SECOND_POSITIVE, '?');
+    }
+
+    @Test
+    public void addsOk() {
+        assertEquals(18, calculator.calculate(FIRST_POSITIVE, SECOND_POSITIVE, '+'), DELTA);
+        assertEquals(3, calculator.calculate(FIRST_NEGATIVE, SECOND_POSITIVE, '+'), DELTA);
+        assertEquals(8, calculator.calculate(FIRST_POSITIVE, SECOND_NEGATIVE, '+'), DELTA);
+        assertEquals(-7, calculator.calculate(FIRST_NEGATIVE, SECOND_NEGATIVE, '+'), DELTA);
+        assertEquals(6, calculator.calculate(0, SECOND_POSITIVE, '+'), DELTA);
+        assertEquals(12, calculator.calculate(FIRST_POSITIVE, 0, '+'), DELTA);
+        assertEquals(0, calculator.calculate(0, 0, '+'), DELTA);
+    }
+
+    @Test
+    public void subtractOK() {
```

Comment:
```suggestion
    public void subtractOk() {
```
check other method names as well